src/decoder/plugins/OpusDecoderPlugin.cxx

```diff
@@ -327,9 +327,15 @@ MPDOpusDecoder::HandleAudio(const ogg_packet &packet)
 			return;
 	}
 
-	// the size of the opus packet in bits / number of samples * samp/sec / 1000
+	/* Formula for calculation of bitrate of the current opus packet:
+	* bits_sent_into_decoder = packet.bytes * 8
+	* bytes_of_audio_decoded = nframes * frame_size
+	* samples_of_audio_decoded_per_channel = bytes_of_audio_decoded / channels / 2
+	* 1/seconds_decoded = opus_sample_rate / samples_of_audio_decoded_per_channel
+	* kbits = bits_sent_into_decoder * 1/seconds_decoded / 1000
+	*/
 	int channels = opus_packet_get_nb_channels((const unsigned char*)packet.packet);
-	uint16_t kbits = (unsigned int)packet.bytes*channels*8 * opus_sample_rate / (nframes * frame_size / 2) / 1000;
+	uint16_t kbits = (unsigned int)packet.bytes*8 * opus_sample_rate / (nframes * frame_size / channels / 2) / 1000;
```

Comment:
Do you realize that `frame_size / channels / 2 == 1`?
`frame_size = channels * frame_size`, see:
https://github.com/MusicPlayerDaemon/MPD/blob/407fa2720a1beeda5f1ea6dd7a995f140cffe0de/src/decoder/plugins/OpusDecoderPlugin.cxx#L241
https://github.com/MusicPlayerDaemon/MPD/blob/407fa2720a1beeda5f1ea6dd7a995f140cffe0de/src/pcm/AudioFormat.hxx#L216-L219

... and you hard-code the sample size to `2`, so what you're effectively doing is `frame_size / frame_size`.

You want the number of frames, but you already have it (`nframes`), so you use it to calculate the uncompressed size (using the frame size), and then divide it by the frame size to get back to the number of frames. Duh!

Reply:
Ohhhh, duh! I had not realized that, and that was the source of my confusion. I get it now, thank you and I apologize. Fixed commit coming right up... that really does make it a lot simpler.